src/Microsoft.AspNetCore.SignalR.Common/Protocol/HubProtocolExtensions.cs

```diff
@@ -29,5 +29,10 @@ public static byte[] GetMessageBytes(this IHubProtocol hubProtocol, HubMessage m
                 MemoryBufferWriter.Return(writer);
             }
         }
+
+        public static int GetMinorVersion(this IHubProtocol protocol)
+        {
+            return (protocol is IHubProtocol2) ? ((IHubProtocol2)protocol).MinorVersion : 0;
```

Comment:
You can make this even sleeker:

```csharp
return (protocol is IHubProtocol2 protocol2) ? protocol2.MinorVersion : 0;
```

The `is` statement can create a new variable where the type has been changed without requiring another cast.